aviary/subsystems/propulsion/propulsion_mission.py

```diff
@@ -16,27 +16,30 @@ class PropulsionMission(om.Group):
 
     def initialize(self):
         self.options.declare(
-            'num_nodes',
-            types=int,
-            lower=0
+            'num_nodes', types=int, lower=0
         )
 
         self.options.declare(
             'aviary_options', types=AviaryValues,
-            desc='collection of Aircraft/Mission specific options')
+            desc='collection of Aircraft/Mission specific options'
+        )
+
+        self.options.declare(
+            'engine_models', types=list,
+            desc='list of EngineModels on aircraft'
+        )
 
     def setup(self):
         nn = self.options['num_nodes']
         options: AviaryValues = self.options['aviary_options']
-        engine_models = options.get_val('engine_models')
+        engine_models = self.options['engine_models']
         num_engine_type = len(engine_models)
 
-        # TODO what if "engine" is not an EngineModel object? Type is never checked/enforced
-
         if num_engine_type > 1:
 
             # We need a single component with scale_factor. Dymos can't find it when it is
             # already sliced across several component.
+            # TODO this only works for engine decks. Need to fix problem in generic way
             comp = om.ExecComp(
                 "y=x",
                 y={'val': np.ones(num_engine_type), 'units': 'unitless'},
```

Comment:
While we are looking here, please add `has_diag_partials=True` so the execomp arguments.